tests/server/test_cors.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+HEADERS_STRANGE_ORIGIN = {'Origin': 'https://example.com'}
+
+
+@pytest.fixture(scope='module')
+def client_defaults():
+    from terracotta.server import create_app
+    flask_app = create_app()
+    with flask_app.test_client() as client:
+        yield client
+
+
+@pytest.fixture(scope='module')
+def client_metadata_disallowed():
+    from terracotta.server import create_app
+    import terracotta
+    terracotta.update_settings(
+        ALLOWED_ORIGINS_METADATA=''
+    )
+    flask_app = create_app()
+    with flask_app.test_client() as client:
+        yield client
+
+
+@pytest.fixture(scope='module')
+def client_all_allowed():
+    from terracotta.server import create_app
+    import terracotta
+    terracotta.update_settings(
+        ALLOWED_ORIGINS_METADATA='*',
+        ALLOWED_ORIGINS_TILES='*'
+    )
+    flask_app = create_app()
+    with flask_app.test_client() as client:
+        yield client
+
+
+@pytest.fixture(scope='module')
+def valid_metadata_path():
+    return '/metadata/val11/x/val12/'
+
+
+@pytest.fixture(scope='module')
+def valid_singleband_path(raster_file_xyz):
+    x, y, z = raster_file_xyz
+    return f'/singleband/val11/x/val12/{z}/{x}/{y}.png'
+
+
+def test_cors_defaults(
+    client_defaults, use_testdb, valid_metadata_path, valid_singleband_path
+):
+    rv = client_defaults.get(valid_metadata_path, headers=HEADERS_STRANGE_ORIGIN)
```

Comment:
What are these headers for?

Reply:
I actually do not need them. The hostname at runtime work just as well. Will remove.